### PR TITLE
document Torque's transient types

### PR DESCRIPTION
**Pull request by tebbi**

### CODE REVIEW

File: src/docs/torque.md [hunk 1]
```diff
@@ -229,6 +229,33 @@ type CompareBuiltinFn = builtin(Context, Object, Object, Object) => Number;
...
+
+```torque
+extern transitioning macro Call(implicit context: Context)
+                               (Callable, Object): Object;
+```
+
+The way this is policed in the type system is that it is illegal to access a value of a transient type across a transitioning operation.
+
+```torque
+const fastArray : FastJSArry = Cast<FastJSArray>(array);
```
Comment (dannoc) [inline src/docs/torque.md]: This code actually doesn't compile, you need the otherwise... not sure how important that is, tho

Comment (mathiasbynens) [inline src/docs/torque.md]: typo: `FastJSArry` --> `FastJSArray`

File: src/docs/torque.md [hunk 2]
```diff
@@ -229,6 +229,33 @@ type CompareBuiltinFn = builtin(Context, Object, Object, Object) => Number;
...
+
+```torque
+// A HeapObject with a JSArray map, and either fast packed elements, or fast
+// holey elements when the global NoElementsProtector is not invalidated.
+transient type FastJSArray extends JSArray
+    generates 'TNode<JSArray>';
+```
+
+For example, in the case of `FastJSArray`, the transient type is invalidated if the array changes to dictionary elements or if the global NoElementsProtector is invalidated.
+The way to express this in Torque is by annotating all callables that could potentially do that as `transitioning`. For example, calling a Javascript function can executy arbitrary Javascript, so it is `transitioning`.
```
Comment (peterwmwong) [inline src/docs/torque.md]: Typo: executy -> execute

**Comment by tebbi** [comment]
Thanks for the reviews everyone!

**Pull request merged**